### PR TITLE
Chore/add tests

### PR DESCRIPTION
**Pull request by alif**
Add linting and `npm test` and CircleCI config.
This is the start to testing for the module.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,212 +1,236 @@
...
+  }
+  return o
+}
 
-module.exports.s3sync = function (config, options) {
-  return new S3Sync(config, options)
+function addToDigest(file) {
+  const md5File = this.prefix + this.hashContents(file)
+  const s3FileName = file.substring(this.path.length)
+  this._digest[s3FileName] = md5File
```
Comment (meeq) [inline index.js]: Since `addToDigest` relies on `this` context to be an `S3Sync` instance, it would make more sense for it to be a class method.

Comment (alif) [reply index.js]: This and `mergeHeaders` were "private" methods.
TBH - I think the class interface has way too many exposed methods so this was an attempt to start moving private methods to truly be private.
I was hoping we could follow up by moving more functions out of the class and only leaving what truly needs to be called from implementations exposed.

Comment (meeq) [reply index.js]: Heads-up, babel now offers a syntax to make class methods private: https://babeljs.io/blog/2018/12/03/7.2.0.html#private-instance-methods-8654-https-githubcom-babel-babel-pull-8654

It is on-track to become standard (currently TC39 Stage 3): https://github.com/tc39/proposal-private-methods

Comment (alif) [reply index.js]: `#` as a prefix is weird
#prefixforprivateisweird

**Pull request merged**